### PR TITLE
Fix some legacy OCaml behaviours

### PR DESCRIPTION
**Pull request by ppedrot**
After the bump to 4.09 tweak some code based comments mentioning the old versions.

### CODE REVIEW

File: ide/coqide/wg_Detachable.ml [hunk 1]
```diff
@@ -54,7 +51,7 @@ class detachable (obj : ([> Gtk.box] as 'a) Gobject.obj) =
       if detached then win#misc#hide ()
       else self#misc#hide ()
 
-    method visible = win#misc#visible || self#misc#visible
+    method visible = win#misc#visible || self#misc#visible [@@ocaml.warning "-7"]
```
Comment (SkySkimmer) [inline ide/coqide/wg_Detachable.ml:54]: Warning 7 is `Method overridden without using the "method!" keyword`, is there a reason we don't use `method!` ?

File: vernac/search.ml [hunk 2]
```diff
@@ -153,18 +153,12 @@ end
...
-  (* use an option to make the function tail recursive. Will be
-     obsoleted with Ocaml 4.02 with the [match … with | exception …]
-     syntax. *)
-  let next = begin
-      try Some (PriorityQueue.maximum q)
-      with Heap.EmptyHeap -> None
-  end in
-  match next with
-  | Some (gref,kind,env,t,_) ->
+  (* Tail-rec! *)
```
Comment (SkySkimmer) [inline vernac/search.ml:156]: We could use the tailrec attribute instead of an unchecked comment.

**Review by SkySkimmer — APPROVED** [review]

File: vernac/search.ml (hunk 2 above)
Comment (ppedrot) [reply vernac/search.ml:156]: I didn't manage to make this work actually, and it's probably not very important either.

**Comment by Alizter** [comment]
@coqbot minimize

**Comment by Alizter** [comment]
@coqbot ci minimize

**Comment by ppedrot** [comment]
I don't think this is related, it compiled fine before. It may be an upstream change instead.

**Comment by Alizter** [comment]
Worst case it actually finds something.

**Pull request merged**